.github/workflows/publish-to-npm.yml

```diff
@@ -1,6 +1,7 @@
 name: npm-publish
 on:
-  push:

```

Comment:
You will also want to follow how we do this in the cli where we only publish on the main branch